tools/swift-inspect/Sources/swift-inspect/DarwinRemoteProcess.swift

```diff
@@ -178,59 +180,115 @@ internal final class DarwinRemoteProcess: RemoteProcess {
 }
 
 extension DarwinRemoteProcess {
-  internal var currentTasks: [(threadID: UInt64, currentTask: swift_addr_t)] {
-    var threadList: UnsafeMutablePointer<thread_t>?
-    var threadCount: mach_msg_type_number_t = 0
+  private class PortList: Sequence {
+    let buffer: UnsafeBufferPointer<mach_port_t>
 
-    let result = task_threads(self.task, &threadList, &threadCount)
-    guard result == KERN_SUCCESS else {
-      print("unable to gather threads for process: \(String(cString: mach_error_string(result))) (0x\(String(result, radix: 16)))")
-      return []
+    init?(task: task_t) {
+      var threadList: UnsafeMutablePointer<mach_port_t>?
+      var threadCount: mach_msg_type_number_t = 0
+
+      let result = task_threads(task, &threadList, &threadCount)
```

Comment:
Is there a formal signature for `task_threads`?  I worry about the type change here without any type of memory rebinding.

Reply:
That's mach for you. The signature is:

```
kern_return_t task_threads(task_inspect_t target_task, thread_act_array_t *act_list, mach_msg_type_number_t *act_listCnt);
```

This one actually isn't too bad. `thread_info` is where it gets really gnarly. These APIs aren't nice to use from Swift. I think I got all the types right though.

Reply:
I don't see how `UnsafeMutablePointer<mach_port_t>` (aka `mach_port_t *`) is the same as `thread_act_array_t`.  Do we not need to rebind the memory?

Reply:
It’s a typedef, everything is a Mach port in the end. 

```
typedef	mach_port_t		thread_act_t;
typedef	thread_act_array_t	thread_act_port_array_t;
```

Reply:
Ah, thanks!  I wish that this was easy to lookup.